pkg/math/voteweighted/threshold.go

```diff
@@ -0,0 +1,87 @@
+package voteweighted
+
+import (
+	"math/big"
+
+	"cosmossdk.io/math"
+)
+
+var (
+	Zero       = big.NewInt(0)
+	Two        = big.NewInt(2)
+	OneMillion = big.NewInt(1_000_000)
+)
+
+// SignsDiffer determines whether newPrice and newPrime move in different directions relative to old.
+func SignsDiffer(old *big.Int, newPrice *big.Int, newPrime *big.Int) bool {
+	newSign := (&big.Int{}).Sub(old, newPrice).Sign()
+	newPrimeSign := (&big.Int{}).Sub(old, newPrime).Sign()
+	return newSign != newPrimeSign
+}
+
+// CalculateRelativeTicks computes the ticks of newPrime relative to the tick size of newPrice.
+func CalculateRelativeTicks(old *big.Int, newPrice *big.Int, newPrime *big.Int, ppm *big.Int) *big.Int {
+	/*
+		1_000_000 * abs(newPrice-newPrime)
+		_____________________________
+				(old*ppm)
+	*/
+
+	// abs(new-newPrime)
+	delta := (&big.Int{}).Abs((&big.Int{}).Sub(newPrice, newPrime))
+	// 1_000_000 * delta
+	oneMillionDelta := (&big.Int{}).Mul(OneMillion, delta)
+	// old * ppm
+	oldPpm := (&big.Int{}).Mul(old, ppm)
+	// returns (delta * 1_000_000) / (old * ppm)
+	return (&big.Int{}).Quo(oneMillionDelta, oldPpm)
+}
+
+// CalculateTicks computes the ticks of newPrice relative to old.
+func CalculateTicks(old *big.Int, newPrice *big.Int, ppm *big.Int) *big.Int {
+	/*
+		1_000_000 * abs(old-newPrice)
+		________________________
+			   (old*ppm)
+	*/
+
+	// abs(old-newPrice)
+	delta := (&big.Int{}).Abs((&big.Int{}).Sub(old, newPrice))
+	// 1_000_000 * delta
+	oneMillionDelta := (&big.Int{}).Mul(OneMillion, delta)
+	// old * ppm
+	oldPpm := (&big.Int{}).Mul(old, ppm)
+	// returns (delta * 1_000_000) / (old * ppm)
+	return (&big.Int{}).Quo(oneMillionDelta, oldPpm)
+}
+
+// ThresholdWeightCalc computes the amount of weight taken into account for the given price update.
+// It follows an algorithm which requires increasing vote price correlation as the percent change in
+// the price increases.
+func ThresholdWeightCalc(
+	currentPrice *big.Int,
+	proposedPrice *big.Int,
+	ppm *big.Int,
+	priceInfo PriceInfo,
```

Comment:
Does priceInfo have the current price for all validators who did not post a price for the considered ticker? Otherwise , the zero price-change assumption for validators who did not vote is not considered?